snapcraft/pluginhandler.py

```diff
@@ -226,15 +241,46 @@ def strip(self, force=False):
         snap_files, snap_dirs = self.migratable_fileset_for('snap')
         _migrate_files(snap_files, snap_dirs, self.stagedir, self.snapdir)
 
-        self.mark_done('strip')
+        state = {
+            'snap-files': snap_files,
+            'snap-directories': snap_dirs
+        }
+
+        self.mark_done('strip', state)
+
+    def clean_strip(self):
+        logger.info('Cleaning up step "strip" for part "{}"'.format(self.name))
+
+        with open(self._step_state_file('strip'), 'r') as f:
+            state = yaml.load(f.read())
+
+        _clean_migrated_files(state['snap-files'], state['snap-directories'],
```

Comment:
Oh right, when writing this I assumed that would imply they were dependent parts, but I forgot that multiple parts can supply the same file as long as the contents don't change.